packages/block-editor/src/store/effects.js

```diff
@@ -80,27 +80,33 @@ export default {
 			return;
 		}
 
-		// We can only merge blocks with similar types
-		// thus, we transform the block to merge first
-		const blockB = getBlock( state, secondBlockClientId );
+		const blockB = getBlock( state, clientIdB );
 		const blockBType = getBlockType( blockB.name );
```

Comment:
This variable is now only used within the `hasSelection` condition, it could wait to be assigned 'til known that the condition is satisfied. A marginal optimization, though potentially the downside that in its current form there's a nice relationship between `blockB` and `blockBType` assignment.